src/common/config.cc

```diff
@@ -1227,6 +1227,49 @@ void md_config_t::diff(
   }
 }
 
+void md_config_t::diff_setting(const md_config_t *other, 
+            map<string, pair<string, string>> *diff, 
+            const string& ceph_setting, bool show_unchanged)
+{
+  Mutex::Locker l(lock);
+
+  char local_buf[4096];
+  char other_buf[4096];
+  for (int i = 0; i < NUM_CONFIG_OPTIONS; i++) {
+    config_option *opt = &config_optionsp[i];
+    std::string ceph_setting_name(opt->name);
+    if (ceph_setting == ceph_setting_name) {
+      memset(local_buf, 0, sizeof(local_buf));
+      memset(other_buf, 0, sizeof(other_buf));
+
+      char *other_val = other_buf;
+      int err = other->get_val(opt->name, &other_val, sizeof(other_buf));
+      if (err < 0) {
+        continue;
+      } 
+
+      char *local_val = local_buf;
+      err = _get_val(opt->name, &local_val, sizeof(local_buf));
+      if (err != 0) {
+        continue;
+      } 
+      if (strcmp(local_val, other_val)) {
+        diff->insert(make_pair(opt->name, make_pair(local_val, other_val)));
```

Comment:
I understand it, but the `map` uses it, not only as a key and it serves as a wrapper also for the `pair`. That's what allows us to use the *iterators*.